core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/general/ClassLeakingRule.java

```diff
@@ -147,7 +153,7 @@ public Map<String, EventAvailability> getRequiredEvents() {
 	private IResult getResult(
 		IItemCollection items, IPreferenceValueProvider valueProvider, IResultValueProvider dependencyResults) {
 		int warningLimit = (int) valueProvider.getPreferenceValue(WARNING_LIMIT).longValue();
-
+		CONFIGURED_TIMEOUT = (int) valueProvider.getPreferenceValue(MAX_TIMEOUT).longValue();
```

Comment:
Nit: This preference should use .clampedLongValueIn and passing in the appropriate unit, so since the timeout use of this specifies TimeUnit.MINUTES we should use valueProvider.getPreferenceValue(MAX_TIMEOUT).clampedLongValueIn(UnitLookup.MINUTE).longValue().

Reply:
If we use clampedLongValueIn then longValue() should be removed right? We can't use both together.

Reply:
Yes, sorry about that confusion.